libkv.go

```diff
@@ -30,29 +33,98 @@ func NewLibKV(url *url.URL) *LibKV {
 
 	sourceType = ""
 	client = ""
+	options := store.Config{}
 
 	if url.Scheme == "consul" {
 		consul.Register()
 		sourceType = store.CONSUL
 		client = env.Getenv("CONSUL_HTTP_ADDR", "localhost:8500")
+		if timeout := env.Getenv("CONSUL_TIMEOUT", ""); timeout != "" {
+			num, err := strconv.ParseInt(timeout, 10, 16)
+			if err != nil {
+				logFatal("consul timeout error", err)
+			}
+			options.ConnectionTimeout = time.Duration(num) * time.Second
+		}
+		if ssl := env.Getenv("CONSUL_HTTP_SSL", ""); ssl != "" {
```

Comment:
Unfortunately CONSUL_HTTP_ADDR doesn't specify if it is TLS or not. Instead there is a separate CONSUL_HTTP_SSL option which needs setting to true if using TLS.

I'll look at adding consul+http / consul+https as additional options.